### PR TITLE
python3Packages.nuitka: 1.1.5 -> 1.8.4

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

update to the latest stable release and gain support for python3.11

changelog:
https://nuitka.net/posts/nuitka-release-18.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by robert-manchester** [comment]
@ofborg build python310.pkgs.nuitka
@ofborg build python311.pkgs.nuitka

**Comment by natsukium** [comment]
Could you also update the format, nativeBuildInputs, propagatedBuildInputs?
I would like to enable pythonImportsCheck and checkPhase.
I could not get the test passed on darwin.

related PRs
https://github.com/NixOS/nixpkgs/pull/242426
https://github.com/NixOS/nixpkgs/pull/253880

**Comment by robert-manchester** [comment]
> Could you also update the format, nativeBuildInputs, propagatedBuildInputs? I would like to enable pythonImportsCheck and checkPhase. I could not get the test passed on darwin.
> 
> related PRs #242426 #253880

according to the tests, nuitka is slow. lots of lines like:
```
CPython took 0.01s vs 1.53s Nuitka.
```
but tests do pass on x64 linux and #263242 builds and runs. no idea about darwin.

File: pkgs/development/python-modules/nuitka/default.nix [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
     rev = version;
-    hash = "sha256-8eWOcxATVS866nlN39b2VU1CuXAfcn0yQsDweHS2yDU=";
+    hash = "sha256-spa3V9KEjqmwnHSuxLLIu9hJk5PrRwNyOw72sfxBVKo=";
   };
 
-  nativeCheckInputs = [ vmprof ];
-  nativeBuildInputs = [ scons ];
-  propagatedBuildInputs = [ chrpath ];
+  nativeCheckInputs = [ ccache ordered-set ];
+  nativeBuildInputs = [ setuptools ];
```
Comment (natsukium) [inline pkgs/development/python-modules/nuitka/default.nix:24]: `ordered-set` should go to `propagatedBuildInputs` and we also need to add `zstandard`.

https://github.com/Nuitka/Nuitka/blob/5a2d18f027031a9b1b39e81ff3c03e08f0ca172b/setup.py#L360-L370

**Review by natsukium — COMMENTED** [review]
Thanks for adopting my suggestions.

If you can't fix the darwin build, you can mark it as broken since I don't think it works on darwin.

File: pkgs/development/python-modules/nuitka/default.nix (hunk 1 above)
Comment (robert-manchester) [reply pkgs/development/python-modules/nuitka/default.nix:24]: i didn't think that they should go into the propagatedBuildInputs as they are just build time and not runtime dependencies and there is no propagatedNativeBuildInputs list so figured it is just up to the caller to add them to the native inputs, if needed.

**Comment by robert-manchester** [comment]
> Thanks for adopting my suggestions.
> 
> If you can't fix the darwin build, you can mark it as broken since I don't think it works on darwin.

darwin fails with linker errors -- marked as broken.

File: pkgs/development/python-modules/nuitka/default.nix (hunk 1 above)
Comment (robert-manchester) [reply pkgs/development/python-modules/nuitka/default.nix]: ok -- i was confused. ordered-set and zstandard are not showing up in the references of packages that use nuitka. change is updated with your suggestion.

File: pkgs/development/python-modules/nuitka/default.nix [hunk 2]
```diff
@@ -1,48 +1,51 @@
...
-  doCheck = false;
+  pythonImportsCheck = [ "nuitka" ];
 
   # Requires CPython
   disabled = isPyPy;
 
+  # tests fail with linker errors on darwin
+  broken = stdenv.isDarwin;
+
   meta = with lib; {
```
Comment (natsukium) [inline pkgs/development/python-modules/nuitka/default.nix:49]:
```suggestion
  meta = with lib; {
    # tests fail with linker errors on darwin
    broken = stdenv.isDarwin;
```
Besides this, LGTM

Comment (robert-manchester) [reply pkgs/development/python-modules/nuitka/default.nix]: fixed

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**